src/failures/src/communication/writers/transaction-writer.ts

```diff
@@ -0,0 +1,107 @@
+import { Reader, Updatable, Writer } from '..';
+
+/**
+ * Provides transactional writing on top of another writer.
+ *
+ * Requires that the written variable is read by a transactional reader.
+ */
+export class TransactionWriter implements Writer {
+    private retryAfterNumberOfUpdates: number;
+
+    private valueSimVar: Writer & Updatable;
+
+    private transactionSimVar: Reader & Writer;
+
+    private openTransactions: Record<number, Transaction> = {};
+
+    /**
+     *
+     * @param valueSimVar The writer to use for writing values.
+     * @param transactionSimVar The reader/writer to use for managing the transaction.
+     * @param retryAfterNumberOfUpdates The number of calls to `update` to wait before making another write attempt. Defaults to 30.
+     */
+    constructor(valueSimVar: Writer & Updatable, transactionSimVar: Reader & Writer, retryAfterNumberOfUpdates?: number) {
+        this.retryAfterNumberOfUpdates = retryAfterNumberOfUpdates || 30;
```

Comment:
```suggestion
    constructor(valueSimVar: Writer & Updatable, transactionSimVar: Reader & Writer, retryAfterNumberOfUpdates: number = 30) {
        this.retryAfterNumberOfUpdates = retryAfterNumberOfUpdates;
```

This is better because || is a falsy-check - so passing 0 would be automatically replaced with 30.

An alternative would be to explicitly check for `=== undefined` (or `typeof x ==="undefined"` if you're really paranoid).